packages/modules/pricing/src/repositories/pricing.ts

```diff
@@ -19,12 +19,46 @@ export class PricingRepository
   extends MikroOrmBase
   implements PricingRepositoryService
 {
+  #availableAttributes: Set<string> = new Set()
+
   constructor() {
     // @ts-ignore
     // eslint-disable-next-line prefer-rest-params
     super(...arguments)
   }
 
+  clearAvailableAttributes() {
+    this.#availableAttributes.clear()
+  }
+
+  async #cacheAvailableAttributes() {
+    const manager = this.getActiveManager<SqlEntityManager>()
+    const knex = manager.getKnex()
+
+    const { rows } = await knex.raw(
+      `
+      SELECT DISTINCT attribute 
+      FROM (
+        SELECT attribute 
+        FROM price_rule 
+        UNION ALL
+        SELECT attribute 
+        FROM price_list_rule
+      ) as combined_rules_attributes
+    `
+    )
+    this.#availableAttributes.clear()
+    rows.forEach(({ attribute }) => {
+      this.#availableAttributes.add(attribute)
+    })
+  }
+
+  async #getCachedAvailableAttributes() {
+    if (this.#availableAttributes.size === 0) {
+      await this.#cacheAvailableAttributes()
+    }
+  }
```

Comment:
super-nit(non-blocking): This is very nitpicky, but the name here indicates a return value. I would either return the attributes or rename them to `checkForCachedAvailableAttributes`

Also, feel free to discard this, just a thought to improve readability for the next dev coming by this :)

Reply:
done, I had the same feeling :)